crates/spirv-std/src/arch.rs

```diff
@@ -188,6 +188,7 @@ pub unsafe fn read_clock_uvec2_khr<V: Vector<u32, 2>, const SCOPE: u32>() -> V {
 }
 
 #[spirv_std_macros::gpu_only]
+#[allow(dead_code)]
```

Comment:
~~This *seems* like a bug? You could use `#[expect(dead_code)]` instead, so that we get an error if it does get fixed, e.g.: https://godbolt.org/z/sdTb3neqz (you won't need the `#[deny(unfulfilled_lint_expectations)]` since we have `#![deny(warnings)]` IIRC)~~

Or maybe it's only linting when compiled for the CPU, because of `#[spirv_std_macros::gpu_only]` (which replaces the body of the function with an unconditional panic).

In that case, this might be fixable by keeping `#[spirv_std_macros::gpu_only]` on `call_glsl_op_with_ints` but removing it from all callers of `call_glsl_op_with_ints` (it's redundant anyway, they just dispatch to `call_glsl_op_with_ints`).

If we want to keep the `#[spirv_std_macros::gpu_only]` on *only* the exported functions (e.g. if the panic mentions the function name), then I think we can use `#[cfg(target_arch = "spirv")]` instead of `#[spirv_std_macros::gpu_only]` on `call_glsl_op_with_ints`.

In fact that sounds simpler, I am going to suggest it now.

Reply:
> (which replaces the body of the function with an unconditional panic).
Aah, I thought the function would be removed altogether. So the warning makes sense. But why do we start hearing about it now?